app/feature/feature-payments/src/main/kotlin/com/hedvig/android/feature/payments2/details/PaymentDetailsDestination.kt

```diff
@@ -0,0 +1,260 @@
+              text = memberCharge.netAmount.toString(),
+              textAlign = TextAlign.End,
+            )
+          }
+        },
+        modifier = Modifier.padding(vertical = 16.dp),
+      )
+      Divider()
+
+      HorizontalItemsWithMaximumSpaceTaken(
+        startSlot = {
+          Text(stringResource(id = R.string.PAYMENTS_PAYMENT_DUE))
+        },
+        endSlot = {
+          Text(
+            text = dateTimeFormatter.format(memberCharge.dueDate.toJavaLocalDate()),
+            textAlign = TextAlign.End,
+            modifier = Modifier.fillMaxWidth(),
+            color = MaterialTheme.colorScheme.onSurfaceVariant,
+          )
+        },
+        modifier = Modifier.padding(vertical = 16.dp),
+      )
+
+      when (memberCharge.status) {
+        MemberCharge.MemberChargeStatus.UPCOMING -> PaymentStatusCard(
+          text = stringResource(id = R.string.PAYMENTS_UPCOMING_PAYMENT),
+          icon = Icons.Hedvig.InfoFilled,
+          iconColor = MaterialTheme.colorScheme.infoElement,
+          colors = CardDefaults.outlinedCardColors(
+            containerColor = MaterialTheme.colorScheme.infoContainer,
+            contentColor = MaterialTheme.colorScheme.onInfoContainer,
+          ),
+          underTextContent = null,
+        )
+
+        MemberCharge.MemberChargeStatus.SUCCESS -> PaymentStatusCard(
+          text = stringResource(id = R.string.PAYMENTS_PAYMENT_SUCCESSFUL),
+          icon = Icons.Hedvig.Checkmark,
+          iconColor = MaterialTheme.colorScheme.onTypeContainer,
+          colors = CardDefaults.outlinedCardColors(
+            containerColor = MaterialTheme.colorScheme.typeContainer,
+            contentColor = MaterialTheme.colorScheme.onTypeContainer,
+          ),
+          underTextContent = null,
+        )
+
+        MemberCharge.MemberChargeStatus.PENDING -> PaymentStatusCard(
+          text = stringResource(id = R.string.PAYMENTS_IN_PROGRESS),
+          icon = Icons.Hedvig.InfoFilled,
+          iconColor = MaterialTheme.colorScheme.infoElement,
+          colors = CardDefaults.outlinedCardColors(
+            containerColor = MaterialTheme.colorScheme.infoContainer,
+            contentColor = MaterialTheme.colorScheme.onInfoContainer,
+          ),
+          underTextContent = null,
+        )
+
+        MemberCharge.MemberChargeStatus.FAILED -> PaymentStatusCard(
+          text = stringResource(
+            id = R.string.PAYMENTS_PAYMENT_FAILED, "-",
+          ),
+          icon = Icons.Hedvig.WarningFilled,
+          iconColor = MaterialTheme.colorScheme.error,
+          colors = CardDefaults.outlinedCardColors(
+            containerColor = MaterialTheme.colorScheme.errorContainer,
+            contentColor = MaterialTheme.colorScheme.onErrorContainer,
+          ),
+          underTextContent = null,
+        )
+
+        MemberCharge.MemberChargeStatus.UNKNOWN -> {}
+      }
+
+      Spacer(Modifier.height(32.dp))
+
+      paymentConnection?.connectionInfo?.let {
```

Comment:
If paymentConncetion is null, so the last item here is just the spacer above, it will mean that we'll have the 32.dp space anyway + the 16.dp space from the padding applied to the outer column, making this have 48.dp of space at the bottom right?
Could we only apply this space inside the let, so that it only gets added if it's needed?

Always tricky with such conditional UI + spacings of course, this one just caught my eye because it might feel like way too much spacing at the bottom